dbms/src/IO/BrotliWriteBuffer.cpp

```diff
@@ -0,0 +1,119 @@
+#include <Common/config.h>
+#if USE_BROTLI
+
+#include <IO/BrotliWriteBuffer.h>
+#include <brotli/encode.h>
+
+namespace DB
+{
+
+class BrotliWriteBuffer::BrotliStateWrapper
+{
+public:
+    BrotliStateWrapper()
+    : state(BrotliEncoderCreateInstance(nullptr, nullptr, nullptr))
+    {
+    }
+
+    ~BrotliStateWrapper()
+    {
+        BrotliEncoderDestroyInstance(state);
+    }
+
+public:
+    BrotliEncoderState * state;
+};
+
+BrotliWriteBuffer::BrotliWriteBuffer(WriteBuffer & out_, int compression_level, size_t buf_size, char * existing_memory, size_t alignment)
+        : BufferWithOwnMemory<WriteBuffer>(buf_size, existing_memory, alignment)
+        , brotli(new BrotliStateWrapper())
```

Comment:
`std::make_unique` will look more safe.

Reply:
I dont understand why its more safer, but i fixed it

Reply:
If you read code and see `std::make_unique` you are sure that unique_ptr is created. When you see `new`, you have to pause reading and check it.

Another case is when the function is called with two arguments:
`f(new A, new B)` will lead to memory leak in case of exception; `f(std::make_unique<A>(), std::make_unique<B>())` won't.